src/components/forms/profile/user-profile-form.tsx

```diff
@@ -15,14 +17,41 @@ import { fetchJson } from "@src/utils/fetch-json";
 
 import "react-datepicker/dist/react-datepicker.css";
 
+type ProfilePayload = Omit<Profile, "id" | "sub" | "updated_at">;
+
+async function updateOrCreateProfile(
+  router: NextRouter,
+  profilePayload: ProfilePayload,
+  isCreation?: boolean,
+): Promise<void> {
+  const url = "/api/profile/user-profile";
+  const method = isCreation ? "POST" : "PATCH";
+  console.log({ method });
+
+  return fetchJson(url, method, profilePayload).then(async (response) => {
+    const parsedResponse = await response.json();
+
+    if (
+      response.status === HttpStatus.CREATED ||
+      response.status === HttpStatus.OK
+    ) {
+      mutate("/api/profile/user-profile", parsedResponse);
```

Comment:
This is it.
And it works well, no call made and no more "glitches" or "flipping" with the data of the profile.